### PR TITLE
Changes to qubit object from Starmon branch

### PR DESCRIPTION
**Pull request by AdriaanRol**
Relates to issues #240 , #42 
Changes proposed in this pull request:
- Calibration routines all exist in qubit abstract base class (this should make time domain calibration much easier). 
- Add plenty of docstrings and other comments 
- Added some characterization parameters to the qubit (e.g., T1, T2echo etc).

### CODE REVIEW

**Comment by AdriaanRol** [comment]
@fluthi @elrama- @NielsBultink @brianzi 
Could any of you take a look at this PR? I would like this to be up to date as a starting point for the M18 milestone rush.

File: pycqed/instrument_drivers/meta_instrument/qubit_objects/qubit_object.py [hunk 1]
```diff
@@ -343,45 +529,16 @@ def find_frequency(self, method='spectroscopy', pulsed=False,
...
-                    # If the last step is > T2* then the next will be for sure
-                    if verbose:
-                        print('Breaking of measurement because of T2*')
-                    break
-            if verbose:
-                print('Converged to: {:.9e}'.format(cur_freq))
-            if update:
-                self.f_qubit.set(cur_freq)
-            return cur_freq
+    def find_frequency_pulsed():
```
Comment (brianzi) [inline pycqed/instrument_drivers/meta_instrument/qubit_objects/qubit_object.py]: raise NotImplementedError?

File: pycqed/instrument_drivers/meta_instrument/qubit_objects/qubit_object.py [hunk 2]
```diff
@@ -343,45 +529,16 @@ def find_frequency(self, method='spectroscopy', pulsed=False,
...
+        pass
+
+    def find_frequency_cw_spec():
+        pass
```
Comment (brianzi) [inline pycqed/instrument_drivers/meta_instrument/qubit_objects/qubit_object.py]: not implemented error

File: pycqed/instrument_drivers/meta_instrument/qubit_objects/qubit_object.py [hunk 3]
```diff
@@ -410,6 +567,86 @@ def find_resonator_frequency(self, use_min=False,
...
+            self.Q_amp180.set(ampl)
+        return ampl
+
+    def calibrate_pulse_amplitude_flipping(self,
+                                           MC=None, update=True,
+                                           fine_accuracy=0.005,
+                                           desired_accuracy=0.00005,
+                                           max_iterations=10,
+                                           verbose=True):
+
```
Comment (brianzi) [inline pycqed/instrument_drivers/meta_instrument/qubit_objects/qubit_object.py]: docstring

**Comment by AdriaanRol** [comment]
@brianzi I've addressed your comments, let me know if this is OK.

**Pull request merged**